File: Anul II/Semestrul II/Medii de proiectare si programare/AgentiiZboruriC/domain/Turist.cs
namespace AgentiiZboruriC.domain;

public class Turist : Persoana
{
    public int? NrClient { get; set; }

    public Turist(int id, string nume, string prenume, int? nrClient) : base(id, nume, prenume)
    {
        this.NrClient = nrClient;
    }

    public override string ToString()
    {
        return $"Turist{{nrClient={NrClient}}}";
    }

    public override bool Equals(object obj)
    {
        if (this == obj) return true;
        if (obj == null || GetType() != obj.GetType()) return false;

        Turist turist = (Turist)obj;

        return NrClient != null ? NrClient.Equals(turist.NrClient) : turist.NrClient == null;
    }

    public override int GetHashCode()
    {
        return NrClient != null ? NrClient.GetHashCode() : 0;
    }
}

File: Anul II/Semestrul II/Medii de proiectare si programare/AgentiiZboruri/Model/src/main/java/ro/model/Turist.java
package ro.model;

public class Turist extends Persoana{
    Integer nrClient;

    public Turist(Integer id, String nume, String prenume, Integer nrClient) {
        super(id, nume, prenume);
        this.nrClient = nrClient;
    }

    public Integer getNrClient() {
        return nrClient;
    }

    public void setNrClient(Integer nrClient) {
        this.nrClient = nrClient;
    }

    @Override
    public String toString() {
        return "Turist{" +
                "nrClient=" + nrClient +
                '}';
    }

    @Override
    public Integer getId() {
        return super.getId();
    }

    @Override
    public void setId(Integer id) {
        super.setId(id);
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Turist turist = (Turist) o;

        return nrClient != null ? nrClient.equals(turist.nrClient) : turist.nrClient == null;
    }

    @Override
    public int hashCode() {
        return nrClient != null ? nrClient.hashCode() : 0;
    }

    public String getNume() {
        return super.getNume();
    }

    public void setNume(String nume) {
        super.setNume(nume);
    }

    public String getPrenume() {
        return super.getPrenume();
    }

    public void setPrenume(String prenume) {
        super.setPrenume(prenume);
    }
}
